sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson

```diff
@@ -1262,5 +1262,21 @@
       stage: V3
       tests: []
     }
+
+    {
+      name: rom_e2e_self_hash
+      desc: '''Verify that the expected ROM has been properly baked into the nelist.
+
+      - Load the contents of the ROM into a buffer.
+      - Hash the buffer with cSHAKE256 (same algorithm used by hardware).
+        - Use empty function string.
+        - Use "ROM_CTRL" customization string.
+      - Verify the hash against golden hash baked into the test.
```

Comment:
since we are checking this against a golden hash, we don't have to use the same algorithm used by hardware. `sha256` should be ok to use here and using the silicon_creator `hmac` driver can help simplify things.

Reply:
This is true. Wasn't sure what people preferred here so just went with the same as HW. We can use SHA256 instead. Both will work with crytpolib. If you feel strongly I can change to SHA256.

Reply:
changed.